compiler/rustc_const_eval/src/const_eval/machine.rs

```diff
@@ -438,10 +418,75 @@ impl<'mir, 'tcx> interpret::Machine<'mir, 'tcx> for CompileTimeInterpreter<'mir,
             }
         }
 
-        // This is a const fn. Call it.
-        // In case of replacement, we return the *original* instance to make backtraces work out
-        // (and we hope this does not confuse the FnAbi checks too much).
-        Ok(Some((ecx.load_mir(instance.def, None)?, orig_instance)))
+        // This is a const fn. Return its mir to be called.
+        ecx.load_mir(instance.def, None).map(Either::Left)
+    }
+
+    #[inline(always)]
+    fn call_extra_fn(
+        ecx: &mut InterpCx<'mir, 'tcx, Self>,
+        fn_val: Self::ExtraFnVal,
+        abis: (CallAbi, &rustc_target::abi::call::FnAbi<'tcx, Ty<'tcx>>),
+        args: &[FnArg<'tcx>],
+        destination: &PlaceTy<'tcx, Self::Provenance>,
+        target: Option<mir::BasicBlock>,
+        unwind: mir::UnwindAction,
+    ) -> InterpResult<'tcx> {
```

Comment:
```suggestion
    ) -> InterpResult<'tcx, Option<Instance<'tcx>>> {
```

then you can return `Ok(None)` when it got handled, and `Ok(Some(instance))` when the normal code path should be taken

Reply:
I just did a different solution (586d704944b039004bb842a78efb4b3a1664d6f3), but I can revert that and do this instead. 🤔